### PR TITLE
Fix leak and crash

### PR DESCRIPTION
**Pull request by mafintosh**

### CODE REVIEW

File: include/udx.h [hunk 1]
```diff
@@ -193,6 +193,8 @@ struct udx_stream {
 
   udx_cirbuf_t outgoing;
   udx_cirbuf_t incoming;
+
+  udx_fifo_t send_queue_other;
```
Comment (kasperisager) [inline include/udx.h:197]: What's the reason for the `_other` postfix?

Comment (mafintosh) [reply include/udx.h:197]: trying to indicate that this is only for outgoing outside the cirbuf, but yea, didn't like it either. ideas?

Comment (mafintosh) [reply include/udx.h:197]: unordered mb?

Comment (kasperisager) [reply include/udx.h:197]: Unordered seems more descriptive 👏🏼

**Comment by kasperisager** [comment]
Also wondering if we should investigate automating test runs with sanitizers enabled at some point, might help catch similar cases in the future.

File: include/udx.h (hunk 1 above)
Comment (mafintosh) [reply include/udx.h:197]: will fix

**Comment by mafintosh** [comment]
@kasperisager 100%

File: include/udx.h (hunk 1 above)
Comment (mafintosh) [reply include/udx.h]: now called `unordered`

**Review by kasperisager — APPROVED** [review]
🦾

**Pull request merged**